src/main/java/com/couchbase/lite/support/Batcher.java

```diff
@@ -161,6 +162,12 @@ public void queueObjects(List<T> objects) {
         }
     }
 
+    public void flushAllAndWait() {
+        synchronized (flushAllMutext) {
+            flushAll(true);
+        }
+    }
+
```

Comment:
> Also can the batcher be flushed without waiting to finish?

Without wait, it causes OOM with large data set (I have used for testing).

> I think it's better to fix this in the flushAll method instead of having another flush method.

sure. I will update it.